### PR TITLE
chore(docs): update security headers specification

### PR DESCRIPTION
**Pull request by j-mendez**
1. fix details on how next handles security headers

* note: check your HSTS at https://hstspreload.org
* if your using headers in the config you need to add them below like 

Screen Shot 2022-01-25 at 9 58 30 PM

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: docs/advanced-features/security-headers.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ This header controls DNS prefetching, allowing browsers to proactively perform d
 
 This header informs browsers it should only be accessed using HTTPS, instead of using HTTP. Using the configuration below, all present and future subdomains will use HTTPS for a `max-age` of 2 years. This blocks access to pages or subdomains that can only be served over HTTP.
 
-If you're deploying to [Vercel](https://vercel.com/docs/edge-network/headers#strict-transport-security), this header is not necessary as it's automatically added to all deployments.
+If you're deploying to [Vercel](https://vercel.com/docs/edge-network/headers#strict-transport-security), this header is not necessary as it's automatically added to all deployments unless you declare headers in your [`next.config.js`](https://nextjs.org/docs/api-reference/next.config.js/introduction)`.
```
Comment (styfle) [inline docs/advanced-features/security-headers.md:46]:
```suggestion
If you're deploying to [Vercel](https://vercel.com/docs/edge-network/headers#strict-transport-security), this header is not necessary as it's automatically added to all deployments unless you declare [`headers`](https://nextjs.org/docs/api-reference/next.config.js/headers) in your [`next.config.js`](https://nextjs.org/docs/api-reference/next.config.js/introduction)`.
```

File: docs/advanced-features/security-headers.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ This header controls DNS prefetching, allowing browsers to proactively perform d
 
 This header informs browsers it should only be accessed using HTTPS, instead of using HTTP. Using the configuration below, all present and future subdomains will use HTTPS for a `max-age` of 2 years. This blocks access to pages or subdomains that can only be served over HTTP.
 
-If you're deploying to [Vercel](https://vercel.com/docs/edge-network/headers#strict-transport-security), this header is not necessary as it's automatically added to all deployments unless you declare headers in your [`next.config.js`](https://nextjs.org/docs/api-reference/next.config.js/introduction)`.
+If you're deploying to [Vercel](https://vercel.com/docs/edge-network/headers#strict-transport-security), this header is not necessary as it's automatically added to all deployments unless you declare [`headers`](https://nextjs.org/docs/api-reference/next.config.js/headers) in your `next.config.js`.
```
Comment (j-mendez) [inline docs/advanced-features/security-headers.md:46]: @styfle  I thought about this at first, but it was essentially a duplicate link for the page.

Comment (j-mendez) [reply docs/advanced-features/security-headers.md:46]: https://www.youtube.com/watch?v=5AsLWIuNNMU old video to my understanding two links for the same page would boost the links page rank for the page priority making this the most important link which may be fine. Appreciate the suggestions on the pr 🙌.

File: docs/advanced-features/security-headers.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ This header controls DNS prefetching, allowing browsers to proactively perform d
 
 This header informs browsers it should only be accessed using HTTPS, instead of using HTTP. Using the configuration below, all present and future subdomains will use HTTPS for a `max-age` of 2 years. This blocks access to pages or subdomains that can only be served over HTTP.
 
-If you're deploying to [Vercel](https://vercel.com/docs/edge-network/headers#strict-transport-security), this header is not necessary as it's automatically added to all deployments.
+If you're deploying to [Vercel](https://vercel.com/docs/edge-network/headers#strict-transport-security), this header is not necessary as it's automatically added to all deployments unless you declare [`headers`](https://nextjs.org/docs/api-reference/next.config.js/headers) in your `next.config.js`.
```
Comment (styfle) [inline docs/advanced-features/security-headers.md:46]:
```suggestion
If you're deploying to [Vercel](https://vercel.com/docs/edge-network/headers#strict-transport-security), this header is not necessary as it's automatically added to all deployments unless you declare [`headers`](/docs/api-reference/next.config.js/headers.md) in your `next.config.js`.
```

**Review by styfle — APPROVED** [review]
Thanks!

File: docs/advanced-features/security-headers.md (hunk 2 above)
Comment (styfle) [reply docs/advanced-features/security-headers.md]: Seems correct to me since this page is about headers afterall 👍

**Pull request merged**